### PR TITLE
Move withApiClient state to this context

### PR DESCRIPTION
**Pull request by coderkevin**
This moves what was held in state for `withApiClient` to the `this`
context. The reasons for this are to manage these updates more carefully
rather than just updating state which causes a re-render.

To Test:
1. `npm install`, `npm test` and ensure all tests pass.
2. `npm run build`, `npm link`
3. cd to `examples/hello-world`, `npm install`, `npm link @fresh-data/framework`, `npm start`
4. cd to `examples/hello-world`, `npm install`, `npm link @fresh-data/framework`, `npm start`

### CODE REVIEW

**Comment by coderkevin** [comment]
I'm not 100% certain this change needs to be made right now, so I'll wait on it for a bit.

File: src/react-redux/with-api-client.js [hunk 1]
```diff
@@ -28,61 +28,82 @@ export default function withApiClient( apiName, options ) {
 			static childContextTypes = WrappedComponent.childContextTypes;
 			static WrappedComponent = WrappedComponent;
 
-			state = { clientKey: null, client: null, clientState: null };
+			constructor( props ) {
+				super( ...arguments );
```
Comment (dmsnell) [inline src/react-redux/with-api-client.js]: is this allowed? seems odd. if using `arguments` I would have expected something like `super.apply( this, arguments )`; if using `...` I would have expected `constructor( ...args ) { super( ...args ); const [ props ] = args;`

maybe it's all fine and I'm just not expecting the right things…

Comment (coderkevin) [reply src/react-redux/with-api-client.js]: Yeah, I picked up this one-line from the Gutenberg source. Seems to work fine there.

**Comment by coderkevin** [comment]
Closing this for now, because it's not really needed with the latest other changes.

**Pull request closed without merging**